src/main/java/com/redhat/devtools/lsp4ij/features/selectionRange/LSPExtendWordSelectionHandler.java

```diff
@@ -0,0 +1,156 @@
+            TextRange selectionTextRange = getTextRange(selectionRange, document);
+            textRanges.addAll(expandToWholeLinesWithBlanks(editorText, selectionTextRange));
+
+            for (SelectionRange parentSelectionRange = selectionRange.getParent();
+                 parentSelectionRange != null;
+                 parentSelectionRange = parentSelectionRange.getParent()) {
+                TextRange parentSelectionTextRange = getTextRange(parentSelectionRange, document);
+                textRanges.addAll(expandToWholeLinesWithBlanks(editorText, parentSelectionTextRange));
+            }
+        }
+        return new ArrayList<>(textRanges);
+    }
+
+    private static @NotNull List<SelectionRange> getSelectionRanges(@NotNull PsiFile file,
+                                                                    @NotNull Document document,
+                                                                    int offset) {
+        if (ProjectIndexingManager.canExecuteLSPFeature(file) != ExecuteLSPFeatureStatus.NOW) {
+            return Collections.emptyList();
+        }
+
+        // Consume LSP 'textDocument/selectionRanges' request
+        LSPSelectionRangeSupport selectionRangeSupport = LSPFileSupport.getSupport(file).getSelectionRangeSupport();
+        if (selectionRangeSupport == null) {
+            return Collections.emptyList();
+        }
+
+        TextDocumentIdentifier textDocumentIdentifier = LSPIJUtils.toTextDocumentIdentifier(file.getVirtualFile());
+
+        Position position = LSPIJUtils.toPosition(offset, document);
+        var params = new LSPSelectionRangeParams(textDocumentIdentifier, Collections.singletonList(position), offset);
+        CompletableFuture<List<SelectionRange>> selectionRangesFuture = selectionRangeSupport.getSelectionRanges(params);
+        try {
+            waitUntilDone(selectionRangesFuture, file);
+        } catch (
+                ProcessCanceledException e) {//Since 2024.2 ProcessCanceledException extends CancellationException so we can't use multicatch to keep backward compatibility
+            //TODO delete block when minimum required version is 2024.2
+            selectionRangeSupport.cancel();
+            return Collections.emptyList();
+        } catch (CancellationException e) {
+            // cancel the LSP requests textDocument/selectionRanges
+            selectionRangeSupport.cancel();
+            return Collections.emptyList();
+        } catch (ExecutionException e) {
+            LOGGER.error("Error while consuming LSP 'textDocument/selectionRanges' request", e);
+            return Collections.emptyList();
+        }
+
+        if (!isDoneNormally(selectionRangesFuture)) {
+            return Collections.emptyList();
+        }
+
+        // textDocument/selectionRanges has been collected correctly, create list of IJ SelectionDescriptor from LSP SelectionRange list
+        return selectionRangesFuture.getNow(null);
```

Comment:
It seems method should never return null, replace with:

`return selectionRangesFuture.getNow(Colletions.emptyList());`